chingu-voyages/v52-tier3-team-38
# Gary/feature/login error handling

# Description
This PR is primarily focused on error handling for the login and signup forms. This is very surface level error handling to just let the user know that there was a problem either signing up or signing in. If there is time we can come back and flesh it out some more.

## Related to the following Jira/Trello issue 
https://trello.com/c/F0fho90n/37-user-login

## Main changes explained:
- Login and Signup server actions are adjusted to return an object containing the error if there was one, as opposed to redirecting to a basic error page with little explanation. This is used to prompt the error message for the Login and Signup forms.
- I added comments for the supabase implementation to give a general idea of what file does what.
- Seed.js was refactored to handle inclusion of dummy data into database if need be.
- DeleteUsers.js was created as a helper function to delete users from the auth table. This is not meant to be a full fleshed api, just a temporary work around to prevent the db from being cluttered from testing.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. To test error handling for the sign up form, try to make an error appear by fields not present / not in the proper format. Those should have bootstrap prompt the user to correct those fields. To make the submission fail, when creating a user have the password be less than 6 characters. Upon submitting a an error should appear above to let the you know that an error occured and that you should try again. If you can think of other ways to make this form fail, please give it a try. 
4. To test error handling for the sign in form, try to enter a wrong password. Upon submission you should be let known that there was a problem with either the username or password and to try again.

## Note:
Address Validation is not set but will be handled at a later time, just not for this PR.
